src/pages/index.js

```diff
@@ -4,54 +4,148 @@ import Section from "../components/Section.js";
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+    // return userData;
+  })
+  .catch((err) => {
+    console.error("Error. The request has failed: ", err);
+  });
+
 const profileModal = new PopupWithForm(".modal-profile", (inputs) => {
-  userInfo.setUserInfo(inputs);
-  profileModal.close();
+  profileModal.changeButtonText("Saving...");
+  api
+    .updateUserInfo(inputs)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        about: res.about,
+        avatar: res.avatar,
+      });
+      modalFormProfile.reset();
+      profileModal.close();
+      profileModal.changeButtonText("Save");
+    })
+    .catch((err) => {
+      console.error("Error. The request has failed: ", err);
+      profileModal.changeButtonText("Save");
+    });
 });
 profileModal.setEventListeners();
 
+const avatarModal = new PopupWithForm(".modal-avatar", (inputs) => {
+  avatarModal.changeButtonText("Saving...");
+  console.log(inputs.avatar);
+
+  api
+    .updateProfilePicture(inputs.avatar)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        about: res.about,
+        avatar: res.avatar,
+      });
+      modalFormAvatar.reset();
+      avatarModal.close();
+      avatarModal.changeButtonText("Save");
+    })
+    .catch((err) => {
+      console.error("Error. The request has failed: ", err);
+      avatarModal.changeButtonText("Save");
+    });
+});
+avatarModal.setEventListeners();
+const modalFormAvatar = document.querySelector(".modal__form-avatar");
+const avatarFormValidator = new FormValidator(config, modalFormAvatar);
+avatarFormValidator.enableValidation();
+document
+  .querySelector(".profile__button-edit-avatar")
+  .addEventListener("click", () => {
+    avatarModal.open();
+    avatarFormValidator.toggleSubmitButton();
+  });
+
 const modalFormProfile = document.querySelector(".modal__form-profile");
 const inputTitle = document.querySelector('[name = "name"]');
-const inputDesc = document.querySelector('[name = "job"]');
+const inputDesc = document.querySelector('[name = "about"]');
 
 const profileFormValidator = new FormValidator(config, modalFormProfile);
 profileFormValidator.enableValidation();
 document
   .querySelector(".profile__button-edit")
   .addEventListener("click", () => {
-    const { userName, userJob } = userInfo.getUserInfo();
+    const { userName, userAbout } = userInfo.getUserInfo();
     inputTitle.value = userName;
-    inputDesc.value = userJob;
+    inputDesc.value = userAbout;
     profileModal.open();
     profileFormValidator.toggleSubmitButton();
   });
 
 const cardModal = new PopupWithForm(".modal-add", (inputs) => {
-  cardSection.addItem(createCard(inputs));
-
-  cardModal.close();
+  cardModal.changeButtonText("Saving...");
+  api
+    .addNewCard(inputs)
+    .then((res) => {
+      cardList.prepend(createCard(res));
+      cardModal.close();
+      cardModal.changeButtonText("Save");
+      modalFormAdd.reset();
```

Comment:
Please, delete it

* You need to have method `close` inside `PopupWithForm` where you need to reset the form according to the task